httpcore/_async/http2.py

```diff
@@ -265,16 +276,21 @@ async def request(
             b"content-length" in seen_headers or b"transfer-encoding" in seen_headers
         )
 
-        await self.send_headers(method, url, headers, has_body, timeout)
-        if has_body:
-            await self.send_body(stream, timeout)
-
-        # Receive the response.
-        status_code, headers = await self.receive_response(timeout)
-        reason_phrase = get_reason_phrase(status_code)
-        stream = AsyncByteStream(
-            aiterator=self.body_iter(timeout), aclose_func=self._response_closed
-        )
+        await self.connection.max_streams_semaphore.acquire()
+        try:
+            await self.send_headers(method, url, headers, has_body, timeout)
+            if has_body:
+                await self.send_body(stream, timeout)
+
+            # Receive the response.
+            status_code, headers = await self.receive_response(timeout)
+            reason_phrase = get_reason_phrase(status_code)
+            stream = AsyncByteStream(
+                aiterator=self.body_iter(timeout), aclose_func=self._response_closed
+            )
+        except:
```

Comment:
We don't actually care, we just want to make sure that we release the semaphore in one of these two cases:

* We fail to return, because some (any) exception was raised.
* We return okay, and the client code fulfils it's contract to call `stream.close()`, which calls into `_response_closed`.

Reply:
Right, in that case we ought to use `Exception` rather than a bare `except`, I think flake8 recommends that, not sure why that's not flagged in the linting.